feat: add civic TR submission examples

korikuzma opened the pull request:
close #7 

~~Currently making a draft PR because I still need to create prognostic/diagnostic examples. @wesleygoar The TR examples are ready for review and all return 204 status code when doing a dry run with the test api.~~

### What we can't put into ClinVar
* We can't represent substitute therapy groups nicely. I did not create a sample submission for this. The solution that CIViC team uses is creating duplicate submissions for each therapy.
* Drug DB identifiers. We can put in our normalized ID, but it would be nice to be able to add a DB identifier.
* Information found in spreadsheets but not API
  * Variant: variation identifiers, alternate designations, URL

### What we can put in ClinVar, but doesn't work with GKS
* Things found in GKS extensions
  * Normalizer information (label and ID). The reason why we do not store the normalized ID/label in the GKS model is because in MetaKB, we use the source provided id/label. A solution for this is to swap our current solution and stick source provided information in the extensions?
  * Indication information. We use this to map CIViC evidence level to AMP/ASCO/CAP tiers.

### GKS lacking classes
* Prognostic / Diagnostic profiles
* CIViC assertions (I think a `Statement` model?)

korikuzma committed:
feat: add civic TR submission examples

korikuzma commented:
@wesleygoar I'll probably wait for your review to submit to test API?

wesleygoar commented on test_submissions/civic/batch.json:
```diff
@@ -0,0 +1,224 @@
+        "variant": [
+          {
+            "hgvs": "ENST00000288602.6:c.1799T>A",
+            "gene": [
```
@korikuzma Is the gene blank or am I missing it?

korikuzma replied:
It's below that line

wesleygoar replied:
is it just the ID? I am not seeing a gene symbol.

korikuzma replied:
Yup, only one is required as seen [here](https://github.com/ncbi/clinvar/blob/2184a4288bf53de66767ef83fdd1b0f7a4145b48/submission_api_schema/submission_apitest_schema.json#L779-L790).

wesleygoar replied:
Gotcha just making sure.

korikuzma commented:
@wesleygoar for batch, I only did evidence items with the same molecular profile. Lmk if this is an issue

wesleygoar commented:
@korikuzma  can we do a second batch submission test as well with multiple genes included?

korikuzma committed:
add another civic batch sub

korikuzma committed:
rename mp13 to mp12

korikuzma committed:
more renaming

korikuzma commented:
> @korikuzma can we do a second batch submission test as well with multiple genes included?

@wesleygoar added

korikuzma commented:
@wesleygoar I'm cleaning up my script if it makes it easier for your review.

korikuzma committed:
add nb for civic tr sub

korikuzma commented on test_submissions/civic/create_civic_therapy_submissions.ipynb:
```diff
@@ -0,0 +1,518 @@
+    "      within_indication_list.append(_within_indication)\n",
+    "\n",
+    "    drug_for_ta = \";\".join(drug_for_ta_list)\n",
+    "    within_indication = all(within_indication_list)\n",
```
@wesleygoar is this correct? Assuming all drugs have to be within indication for combo therapies

wesleygoar replied:
This is generally true though not a hard and fast rule.

korikuzma replied:
@wesleygoar How should I change it?

wesleygoar replied:
@korikuzma Don't change it. This is the proper way to do it.

korikuzma commented:
@wesleygoar added nb. lmk if you have any qs

korikuzma committed:
add dry run test

korikuzma committed:
add prognostic

korikuzma committed:
add civic diagnostic

korikuzma committed:
add readme for civic test submissions

korikuzma commented:
@wesleygoar okay, I think it's officially ready for review. README was added in addition to diagnostic/prognostic (manually created).

korikuzma committed:
add util function

korikuzma commented on test_submissions/civic/create_civic_therapy_submissions.ipynb:
```diff
@@ -0,0 +1,535 @@
+    "    \"localKey\": study_id,\n",
+    "    \"observedIn\": [{\n",
+    "      \"alleleOrigin\": \"somatic\",\n",
+    "      \"affectedStatus\": \"yes\",\n",
```
I'm hardcoding this. I think I should actually be using the `Direction` in the evidence item to determine this? @ahwagner 

```json
"affectedStatus": {
          "type": "string",
          "description": "Indicates whether or not the individual(s) in each observation were affected by the condition for the interpretation",
          "enum": [
            "yes",
            "no",
            "unknown",
            "not provided",
            "not applicable"
          ]
```

ahwagner commented on test_submissions/civic/prognostic.json:
```diff
@@ -0,0 +1,54 @@
+      "conditionSet": {
+        "condition": [
+          {
+            "db": "MONDO",
+            "id": "0005075"
```
Does ClinVar allow us to submit labels with this, or ID only?

korikuzma replied:
ID or name only. Example of trying to include all returns the following error:

```
"{'db': 'MONDO', 'id': '0005233', 'name': 'Lung Non-Small Cell Carcinoma'} is valid under each of {'required': ['name']}, {'required': ['id', 'db']}"
```

ahwagner reviewed (changes requested):
Hrm. This PR shows how we would submit CIViC Evidence Items (this is okay for testing), but from the file the CIViC team provided I think they are planning to submit CIViC Assertions. This PR should reflect that. I have asked the CIViC team to see if they want Evidence Items supported.

We should also be evaluating the `deletes`

ahwagner commented:
> ### What we can put in ClinVar, but doesn't work with GKS
> * Things found in GKS extensions
>   
>   * Normalizer information (label and ID). The reason why we do not store the normalized ID/label in the GKS model is because in MetaKB, we use the source provided id/label. A solution for this is to swap our current solution and stick source provided information in the extensions?
>   * Indication information. We use this to map CIViC evidence level to AMP/ASCO/CAP tiers.

@korikuzma we should use the `mappings` field for this (inherited from [MappableEntity](https://github.com/ga4gh/gks-common/blob/05ba1bffd042122aad01f3be0cca9c23ecd8549f/schema/gks-common/core-source.yaml#L31-L39)) instead of the old descriptor approach.

korikuzma commented:
> > ### What we can put in ClinVar, but doesn't work with GKS
> > 
> > * Things found in GKS extensions
> >   
> >   * Normalizer information (label and ID). The reason why we do not store the normalized ID/label in the GKS model is because in MetaKB, we use the source provided id/label. A solution for this is to swap our current solution and stick source provided information in the extensions?
> >   * Indication information. We use this to map CIViC evidence level to AMP/ASCO/CAP tiers.
> 
> @korikuzma we should use the `mappings` field for this (inherited from [MappableEntity](https://github.com/ga4gh/gks-common/blob/05ba1bffd042122aad01f3be0cca9c23ecd8549f/schema/gks-common/core-source.yaml#L31-L39)) instead of the old descriptor approach.

@ahwagner Ah okay, we already have mappings but I didn't pull from there. I can stick in the normalizer information there. Not sure about the indication data though...

korikuzma commented:
> > > ### What we can put in ClinVar, but doesn't work with GKS
> > > 
> > > * Things found in GKS extensions
> > >   
> > >   * Normalizer information (label and ID). The reason why we do not store the normalized ID/label in the GKS model is because in MetaKB, we use the source provided id/label. A solution for this is to swap our current solution and stick source provided information in the extensions?
> > >   * Indication information. We use this to map CIViC evidence level to AMP/ASCO/CAP tiers.
> > 
> > 
> > @korikuzma we should use the `mappings` field for this (inherited from [MappableEntity](https://github.com/ga4gh/gks-common/blob/05ba1bffd042122aad01f3be0cca9c23ecd8549f/schema/gks-common/core-source.yaml#L31-L39)) instead of the old descriptor approach.
> 
> @ahwagner Ah okay, we already have mappings but I didn't pull from there. I can stick in the normalizer information there. Not sure about the indication data though...

@ahwagner It's also not guaranteed that the `system` will be the same across GKS representations. For example, some people could use a URL whereas others could use a free text string

korikuzma committed:
add civic assertions

korikuzma committed:
add onco assertion + urls

korikuzma commented:
> but from the file the CIViC team provided I think they are planning to submit CIViC Assertions. This PR should reflect that.

That was my bad. I thought CIViC team was testing out the assertions. I included assertion data in the most recent commits

korikuzma committed:
add delete, update readme, add sub actions

korikuzma commented:
> We should also be evaluating the deletes

I added dummy example of deletes. I couldn't find our SCV accession for the single submission that was actually submitted to the test API

korikuzma committed:
new line

wesleygoar commented:
> Hrm. This PR shows how we would submit CIViC Evidence Items (this is okay for testing), but from the file the CIViC team provided I think they are planning to submit CIViC Assertions. This PR should reflect that. I have asked the CIViC team to see if they want Evidence Items supported.
> 
> We should also be evaluating the `deletes`

@ahwagner can you expand on what you mean by this please?

ahwagner commented:
> > Hrm. This PR shows how we would submit CIViC Evidence Items (this is okay for testing), but from the file the CIViC team provided I think they are planning to submit CIViC Assertions. This PR should reflect that. I have asked the CIViC team to see if they want Evidence Items supported.
> > We should also be evaluating the `deletes`
> 
> @ahwagner can you expand on what you mean by this please?

Which part would you like expanded upon? CIViC Assertions vs. Evidence, or the `deletes`? Or both?

wesleygoar commented:
> > > Hrm. This PR shows how we would submit CIViC Evidence Items (this is okay for testing), but from the file the CIViC team provided I think they are planning to submit CIViC Assertions. This PR should reflect that. I have asked the CIViC team to see if they want Evidence Items supported.
> > > We should also be evaluating the `deletes`
> > 
> > 
> > @ahwagner can you expand on what you mean by this please?
> 
> Which part would you like expanded upon? CIViC Assertions vs. Evidence, or the `deletes`? Or both?

@ahwagner the `deletes`

wesleygoar reviewed (approved):
I still haven't received a response from @ahwagner about 'deletes' but otherwise this looks good to me.

korikuzma commented:
> > We should also be evaluating the deletes
> 
> I added dummy example of deletes. I couldn't find our SCV accession for the single submission that was actually submitted to the test API

This was me being blind. I found the corresponding SCV accession and updated

ahwagner commented on test_submissions/civic/assesrtion_onco.json:
```diff
@@ -0,0 +1,101 @@
+    {
+      "recordStatus": "novel",
+      "localID": "civic.mpid:624",
+      "localKey": "civic.AID:55",
```
Is `localKey` here equivalent to `linkingID` in the submission spreadsheet?

korikuzma committed:
typo
